File: src/mock/mockData.ts
/**
 * mockData.ts
 * Données fictives pour le développement / review sur staging.
 * Jamais importées en production (uniquement depuis MockDataBanner + authStore guards).
 */

import type { ServerStatsData, FriendEntry, FriendsResponse, LeaderboardEntry } from '@/api/client'
import type { User } from '@/types'
import type { ServerStatsMap } from '@/store/authStore'
import { getTodayParis } from '@/store/gameStore'

// ─── User ────────────────────────────────────────────────────────────────────

export const MOCK_USER: User = {
  id: 1,
  email: 'demo@guesstoday.fr',
  displayName: 'Demo User',
  avatarUrl: null,
  emailVerified: true,
}

// ─── Stats ───────────────────────────────────────────────────────────────────

const filmStats: ServerStatsData = {
  gamesPlayed: 47,
  wins: 39,
  currentStreak: 8,
  maxStreak: 15,
  distribution: { '1': 12, '2': 14, '3': 8, '4': 3, '5': 2 },
}

const seriesStats: ServerStatsData = {
  gamesPlayed: 23,
  wins: 17,
  currentStreak: 3,
  maxStreak: 7,
  distribution: { '1': 4, '2': 7, '3': 4, '4': 1, '5': 1 },
}

const wikiStats: ServerStatsData = {
  gamesPlayed: 31,
  wins: 22,
  currentStreak: 5,
  maxStreak: 11,
  distribution: { '1': 6, '2': 8, '3': 5, '4': 2, '5': 1 },
}

export const MOCK_SERVER_STATS: ServerStatsMap = {
  film: filmStats,
  series: seriesStats,
  wiki: wikiStats,
}

// ─── Friends ─────────────────────────────────────────────────────────────────

const today = getTodayParis()

export const MOCK_FRIENDS_RESPONSE: FriendsResponse = {
  date: today,
  today,
  myCode: 'DEMO42',
  friends: [
    {
      id: 1,
      displayName: 'Demo User',
      avatarUrl: null,
      streak: 8,
      isMe: true,
      scores: {
        film:   { attemptsUsed: 2, won: true,  completedAt: new Date().toISOString() },
        series: null,
        wiki:   { attemptsUsed: 4, won: false, completedAt: new Date().toISOString() },
      },
    },
    {
      id: 2,
      displayName: 'Alice M.',
      avatarUrl: null,
      streak: 12,
      isMe: false,
      scores: {
        film:   { attemptsUsed: 1, won: true,  completedAt: new Date().toISOString() },
        series: { attemptsUsed: 3, won: true,  completedAt: new Date().toISOString() },
        wiki:   null,
      },
    },
    {
      id: 3,
      displayName: 'Thomas B.',
      avatarUrl: null,
      streak: 4,
      isMe: false,
      scores: {
        film:   { attemptsUsed: 5, won: false, completedAt: new Date().toISOString() },
        series: null,
        wiki:   { attemptsUsed: 2, won: true,  completedAt: new Date().toISOString() },
      },
    },
    {
      id: 4,
      displayName: 'Camille R.',
      avatarUrl: null,
      streak: 21,
      isMe: false,
      scores: {
        film:   { attemptsUsed: 1, won: true, completedAt: new Date().toISOString() },
        series: { attemptsUsed: 1, won: true, completedAt: new Date().toISOString() },
        wiki:   { attemptsUsed: 3, won: true, completedAt: new Date().toISOString() },
      },
    },
    {
      id: 5,
      displayName: 'Jules D.',
      avatarUrl: null,
      streak: 0,
      isMe: false,
      scores: { film: null, series: null, wiki: null },
    },
  ] satisfies FriendEntry[],
  pending: [
    { id: 6, displayName: 'Sophie L.', direction: 'incoming' },
  ],
}

// ─── Leaderboard global ───────────────────────────────────────────────────────

export const MOCK_LEADERBOARD: LeaderboardEntry[] = [
  {
    id: 4, displayName: 'Camille R.', avatarUrl: null, isMe: false, rank: 1,
    totalWins: 87, totalPlayed: 93, winRate: 94, filmWins: 31, seriesWins: 28, wikiWins: 28,
    filmPlayed: 33, seriesPlayed: 30, wikiPlayed: 30, avgAttempts: 1.8, currentStreak: 21, maxStreak: 34,
  },
  {
    id: 2, displayName: 'Alice M.', avatarUrl: null, isMe: false, rank: 2,
    totalWins: 74, totalPlayed: 85, winRate: 87, filmWins: 27, seriesWins: 24, wikiWins: 23,
    filmPlayed: 30, seriesPlayed: 28, wikiPlayed: 27, avgAttempts: 2.1, currentStreak: 12, maxStreak: 19,
  },
  {
    id: 1, displayName: 'Demo User', avatarUrl: null, isMe: true, rank: 3,
    totalWins: 61, totalPlayed: 78, winRate: 78, filmWins: 22, seriesWins: 17, wikiWins: 22,
    filmPlayed: 28, seriesPlayed: 24, wikiPlayed: 26, avgAttempts: 2.6, currentStreak: 8, maxStreak: 15,
  },
  {
    id: 3, displayName: 'Thomas B.', avatarUrl: null, isMe: false, rank: 4,
    totalWins: 48, totalPlayed: 69, winRate: 70, filmWins: 18, seriesWins: 12, wikiWins: 18,
    filmPlayed: 24, seriesPlayed: 22, wikiPlayed: 23, avgAttempts: 3.0, currentStreak: 4, maxStreak: 11,
  },
  {
    id: 5, displayName: 'Jules D.', avatarUrl: null, isMe: false, rank: 5,
    totalWins: 29, totalPlayed: 51, winRate: 57, filmWins: 11, seriesWins: 8, wikiWins: 10,
    filmPlayed: 18, seriesPlayed: 16, wikiPlayed: 17, avgAttempts: 3.4, currentStreak: 0, maxStreak: 6,
  },
]

File: mobile/ios/GuessToday/App/MockData.swift
#if DEBUG || NRT
import Foundation

enum MockData {

    // MARK: - Friends

    static let friendsPayload = FriendsPayload(
        date: todayString(),
        today: todayString(),
        myCode: "LOUIS7",
        friends: [
            FriendScoreEntry(
                id: 1, displayName: "Moi", streak: 12,
                scores: FriendScores(
                    film: DayScoreEntry(attemptsUsed: 2, won: true, completedAt: iso()),
                    series: nil,
                    wiki: DayScoreEntry(attemptsUsed: 4, won: false, completedAt: iso())
                ),
                isMe: true
            ),
            FriendScoreEntry(
                id: 2, displayName: "Alice M.", streak: 7,
                scores: FriendScores(
                    film: DayScoreEntry(attemptsUsed: 1, won: true, completedAt: iso()),
                    series: DayScoreEntry(attemptsUsed: 3, won: true, completedAt: iso()),
                    wiki: nil
                ),
                isMe: false
            ),
            FriendScoreEntry(
                id: 3, displayName: "Thomas B.", streak: 3,
                scores: FriendScores(
                    film: DayScoreEntry(attemptsUsed: 5, won: false, completedAt: iso()),
                    series: nil,
                    wiki: DayScoreEntry(attemptsUsed: 2, won: true, completedAt: iso())
                ),
                isMe: false
            ),
            FriendScoreEntry(
                id: 4, displayName: "Camille R.", streak: 21,
                scores: FriendScores(
                    film: DayScoreEntry(attemptsUsed: 1, won: true, completedAt: iso()),
                    series: DayScoreEntry(attemptsUsed: 1, won: true, completedAt: iso()),
                    wiki: DayScoreEntry(attemptsUsed: 3, won: true, completedAt: iso())
                ),
                isMe: false
            ),
            FriendScoreEntry(
                id: 5, displayName: "Jules D.", streak: 0,
                scores: FriendScores(film: nil, series: nil, wiki: nil),
                isMe: false
            ),
        ],
        pending: [
            PendingFriendEntry(id: 6, displayName: "Sophie L.", direction: "incoming"),
        ]
    )

    // MARK: - Leaderboard

    static let leaderboardPayload = LeaderboardPayload(leaderboard: [
        LeaderboardEntry(id: 4, displayName: "Camille R.", avatarUrl: nil, isMe: false,
                         rank: 1, totalWins: 87, totalPlayed: 90, winRate: 0.97,
                         filmWins: 32, seriesWins: 28, wikiWins: 27,
                         currentStreak: 21, maxStreak: 35),
        LeaderboardEntry(id: 2, displayName: "Alice M.", avatarUrl: nil, isMe: false,
                         rank: 2, totalWins: 64, totalPlayed: 72, winRate: 0.89,
                         filmWins: 28, seriesWins: 20, wikiWins: 16,
                         currentStreak: 7, maxStreak: 14),
        LeaderboardEntry(id: 1, displayName: "Moi", avatarUrl: nil, isMe: true,
                         rank: 3, totalWins: 55, totalPlayed: 65, winRate: 0.85,
                         filmWins: 24, seriesWins: 18, wikiWins: 13,
                         currentStreak: 12, maxStreak: 18),
        LeaderboardEntry(id: 3, displayName: "Thomas B.", avatarUrl: nil, isMe: false,
                         rank: 4, totalWins: 38, totalPlayed: 50, winRate: 0.76,
                         filmWins: 18, seriesWins: 10, wikiWins: 10,
                         currentStreak: 3, maxStreak: 9),
        LeaderboardEntry(id: 5, displayName: "Jules D.", avatarUrl: nil, isMe: false,
                         rank: 5, totalWins: 12, totalPlayed: 20, winRate: 0.60,
                         filmWins: 7, seriesWins: 3, wikiWins: 2,
                         currentStreak: 0, maxStreak: 4),
    ])

    // MARK: - Stats (par jeu)

    static func localStats(for mode: GameMode) -> LocalStats {
        switch mode {
        case .film:
            return LocalStats(gamesPlayed: 65, wins: 55, currentStreak: 12, maxStreak: 18,
                              distribution: ["1": 8, "2": 14, "3": 17, "4": 10, "5": 6, "X": 10], lastWonDate: todayString())
        case .series:
            return LocalStats(gamesPlayed: 48, wins: 38, currentStreak: 5, maxStreak: 14,
                              distribution: ["1": 5, "2": 9, "3": 12, "4": 8, "5": 4, "X": 10], lastWonDate: todayString())
        case .wiki:
            return LocalStats(gamesPlayed: 52, wins: 31, currentStreak: 3, maxStreak: 9,
                              distribution: ["1": 3, "2": 6, "3": 9, "4": 7, "5": 6, "X": 21], lastWonDate: nil)
        }
    }

    // MARK: - Hub (statut du jour par jeu : gagné / en cours / à jouer)

    /// (challengeNumber, outcome, attemptsUsed) pour construire un DailyChallengeStatus fictif.
    static func hubStatus(for mode: GameMode) -> (number: Int, outcome: String?, attemptsUsed: Int) {
        switch mode {
        case .film:   return (142, "won", 2)   // gagné en 2
        case .series: return (138, nil, 1)     // en cours
        case .wiki:   return (96, nil, 0)      // à jouer
        }
    }

    // MARK: - Community stats (Stats du jour)

    static func communityStats(for mode: GameMode) -> CommunityStats {
        switch mode {
        case .film:   return CommunityStats(totalGames: 1240, totalWins: 892, totalLosses: 348, winRate: 72,
                                            winsByAttempt: ["1": 110, "2": 240, "3": 290, "4": 160, "5": 92])
        case .series: return CommunityStats(totalGames: 980, totalWins: 640, totalLosses: 340, winRate: 65,
                                            winsByAttempt: ["1": 70, "2": 150, "3": 210, "4": 130, "5": 80])
        case .wiki:   return CommunityStats(totalGames: 1105, totalWins: 530, totalLosses: 575, winRate: 48,
                                            winsByAttempt: ["1": 40, "2": 95, "3": 140, "4": 130, "5": 125])
        }
    }

    // MARK: - Helpers

    private static func todayString() -> String {
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd"
        return f.string(from: Date())
    }

    private static func iso() -> String {
        ISO8601DateFormatter().string(from: Date())
    }
}
#endif
